File: python_library/sdc.py
import requests

class SDCar:
    def __init__(self, ip, port):
        self.ip = ip
        self.port = port
        self.left = 0
        self.right = 0
    def getStatus(self):
        return "SDCar: " + self.ip + ":" + str(self.port)
    def getMotorStatus(self):
        return {"left": self.left, "right":  self.right}
    def setLeftMotor(self, value):
        req = requests.get("http://" + self.ip + ":" + str(self.port) + "/left/set/" + str(value))
        self.left = req.json()["value"]
        if req.json()["success"] == True:
            return True
        else:
            return False
    def setRightMotor(self, value):
        req = requests.get("http://" + self.ip + ":" + str(self.port) + "/right/set/" + str(value))
        self.right = req.json()["value"]
        if req.json()["success"] == True:
            return True
        else:
            return False
    def brake(self):
        req = requests.get("http://" + self.ip + ":" + str(self.port) + "/all/brake")
        if req.json()["success"] == True:
            return True
        else:
            return False
    def forward(self, value):
        req = requests.get("http://" + self.ip + ":" + str(self.port) + "/all/forward/" + str(value))
        if req.json()["success"] == True:
            return True
        else:
            return False

File: javascript_library/sdc.js
// THIS VERSION IS MADE FOR A WEB BROWSER

class SDCar {
    constructor(ip, port) {
        this.ip = ip;
        this.port = port;
        this.left = 0
        this.right = 0
    }
    getStatus() {
        return "SDCar: " + this.ip + ":" + this.port
    }
    getMotorStatus() {
        return {"left": self.left, "right":  self.right}
    }
    setLeftMotor(speed) {
        fetch("http://" + this.ip + ":" + this.port + "/left/set/" + speed)
        .then(response => response.json())
        .then(data => {
            if (data.success == true) {
                this.left = speed
                return true
            } else {
                return false
            }
        })
    }
    setRightMotor(speed) {
        fetch("http://" + this.ip + ":" + this.port + "/right/set/" + speed)
        .then(response => response.json())
        .then(data => {
            if (data.success == true) {
                this.right = speed
                return true
            } else {
                return false
            }
        })
    }
    brake() {
        fetch("http://" + this.ip + ":" + this.port + "/all/brake")
        .then(response => response.json())
        .then(data => {
            if (data.success == true) {
                this.right = 0
                this.left = 0
                return true
            } else {
                return false
            }
        })
    }
    forward(speed) {
        fetch("http://" + this.ip + ":" + this.port + "/all/forward/" + speed)
        .then(response => response.json())
        .then(data => {
            if (data.success == true) {
                this.right = -speed
                this.left = speed
                return true
            } else {
                return false
            }
        })
    }
}